code/modules/admin/verbs/selectequipment.dm

```diff
@@ -0,0 +1,164 @@
+	ui = SStgui.try_update_ui(user, src, ui)
+	if(!ui)
+		ui = new(user, src, "SelectEquipment", "Select Equipment")
+		ui.open()
+
+/datum/select_equipment/ui_state(mob/user)
+	return GLOB.admin_state
+
+/datum/select_equipment/ui_status(mob/user, datum/ui_state/state)
+	if(QDELETED(target))
+		return UI_CLOSE
+	return ..()
+
+/datum/select_equipment/ui_close(mob/user)
+	clear_human_dummy(dummy_key)
+	qdel(src)
+
+/datum/select_equipment/proc/init_dummy()
+	dummy_key = "selectequipmentUI_[target]"
+	dummy = generate_or_wait_for_human_dummy(dummy_key)
+	var/mob/living/carbon/C = target
+	if(istype(C))
+		C.dna.transfer_identity(dummy)
+		dummy.updateappearance()
+
+	unset_busy_human_dummy(dummy_key)
+	return
+
+/datum/select_equipment/proc/make_outfit_entries(list/L)
+	var/list/entries = list()
+	for(var/path in L)
+		var/datum/outfit/O = path
+		entries[path] = initial(O.name)
+	return sortAssocList(entries)
+
+//GLOB.custom_outfits lists outfit *objects* so we'll need to do some custom handling for it
+/datum/select_equipment/proc/make_custom_outfit_entries(list/L)
+	var/list/entries = list()
+	for(var/datum/outfit/O in L)
+		cached_custom_outfits[O.name] = O
+		entries[O.name] = O.name //it's either this or special handling on the UI side
+	return sortAssocList(entries)
+
+/datum/select_equipment/ui_static_data(mob/user)
+	var/list/data = list()
+	if(!dummy)
+		init_dummy()
+
+	var/datum/preferences/prefs
+	if(target.client)
+		prefs = target.client.prefs
+	var/icon/dummysprite = get_flat_human_icon(null, prefs=prefs, dummy_key = dummy_key, outfit_override = selected_outfit)
+	data["icon64"] = icon2base64(dummysprite)
+
+	if(!cached_outfits)
+		//the assoc keys here will turn into Tabs in the UI, so make sure to name them well
+		cached_outfits = list()
+		cached_outfits["General"] = list(/datum/outfit = "Naked") + make_outfit_entries(subtypesof(/datum/outfit) - typesof(/datum/outfit/job) - typesof(/datum/outfit/plasmaman))
+		cached_outfits["Jobs"] = make_outfit_entries(typesof(/datum/outfit/job))
+		cached_outfits["Plasmamen Outfits"] = make_outfit_entries(typesof(/datum/outfit/plasmaman))
+
+	cached_outfits["Custom"] = list("Click confirm to open the outfit manager" = "Create a custom outfit...") + make_custom_outfit_entries(GLOB.custom_outfits)
+
+	data["outfits"] = cached_outfits
+	data["name"] = target
+
+	return data
+
+
+/datum/select_equipment/proc/resolve_outfit(text)
+	var/path = text2path(text)
+
+	if(ispath(path, /datum/outfit))
+		return new path
+
+	else //don't bail yet - could be a special option or custom outfit
+		var/datum/outfit/custom_outfit = cached_custom_outfits[text]
+		if(istype(custom_outfit))
+			return custom_outfit
+
+
+/datum/select_equipment/ui_act(action, params)
+	if(..())
+		return
+
+	switch(action)
+		if("preview")
+			var/datum/outfit/O = resolve_outfit(params["path"])
+			if(!istype(O))
+				return
+			selected_outfit = O.type //the typepath - not the object
+			update_static_data(user.mob)
```

Comment:
Suggestion:

Here, set `. = TRUE` instead of this call, to signal that ui_data should be sent.

Rename ui_static_data proc to ui_data.

In ui_interact, after the line which opens the ui, `ui.set_autoupdate(FALSE)`